src/python/resize/test_core.py

```diff
@@ -0,0 +1,37 @@
+from pathlib import Path
+
+import pytest
+from resize import resize
+
+
+@pytest.fixture
+def mock_img_path():
+    return (
+        Path.cwd() / "src" / "python" / "resize" / "sample_data" / "sample_image1.png"
+    )
+
+
+def test_resize_percentage(mock_img_path):
+    # Arrange
+    original = resize(path=mock_img_path, percentage=1.0)
+
+    # Act
+    img = resize(
+        path=mock_img_path,
+        percentage=0.5,
+    )
+
+    # Assert
+    assert img.width == original.width / 2
+    assert img.height == original.height / 2
```

Comment:
this compares integers to floats.
```suggestion
    assert img.width == original.width // 2
    assert img.height == original.height // 2
```